pervane/serve.py

```diff
@@ -102,10 +104,22 @@ def str2bool(v):
 parser.add_argument(
     '--debug', type=str2bool, dest='debug', default=False,
     help='Show debug logs')
+parser.add_argument(
+    '--version', action='store_true', dest='version', default=False,
+    help='Show version and exit')
 
 args = parser.parse_args()
+if args.version:
+  # early return if only the version is asked.
+  version_file = str(Path(__file__).resolve().parent / "version.txt")
+  with open(version_file, "r") as version_fh:
+    version = version_fh.read()
+    print('Pervane ', version)
+    sys.exit()
+
 # Log may not show up in non-debug mode.
```

Comment:
Let's take this before version printing for better debuggability.

Reply:
then, we get the following output for `pervane --version` which I am not sure of
```
$ python3 pervane/serve.py --version
loaded args %s Namespace(allow_multi_user=False, cache_seconds=2, debug=False, front_page_message='Welcome to Pervane! This is default welcome message.', host='0.0.0.0', ignore_patterns=['env/.*', '.git', '.*.swp', '.*.pyc', '__pycache__', '.allmark', '_vnote.json', '!!!meta.json', 'PaxHeader', '.nojekyll', '.*.sqlite'], note_extensions=['.txt', '.md', '.markdown'], password=None, port='5000', root_dir='/home/basri/dev/personal/repos/notes', username=None, version=True)
Pervane  0.0.52
```

Reply:
nvm